public-api/src/main/java/org/pmiops/workbench/privateWorkbench/PrivateWorkbenchConfig.java

```diff
@@ -0,0 +1,35 @@
+package org.pmiops.workbench.privateWorkbench;
+
+import org.pmiops.workbench.config.WorkbenchConfig;
+import org.pmiops.workbench.privateWorkbench.api.ProfileApi;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ScopedProxyMode;
+import org.springframework.http.HttpHeaders;
+import org.springframework.web.context.annotation.RequestScope;
+
+import javax.servlet.http.HttpServletRequest;
+
+@org.springframework.context.annotation.Configuration
+public class PrivateWorkbenchConfig {
+
+  private static final String END_USER_API_CLIENT = "endUserApiClient";
+
+  @Bean(name=END_USER_API_CLIENT)
+  @RequestScope(proxyMode = ScopedProxyMode.DEFAULT)
+  public ApiClient workbenchApiClient(WorkbenchConfig workbenchConfig) {
```

Comment:
Did you check whether the initial approach you were using worked (injecting request here)? I do like that one best as it most closely resembles our similar usages in Workbench.

Reply:
Will try it.